### ISSUE TITLE
Missing input files for rule nohit_annotate_reads

### ISSUE THREAD

**Issue by biojiangke** [WatsonLab/KOunt#4]
After installing and downloading the test data set, I ran:

snakemake -k --ri --use-conda --cores 8

And got a missing input error at nohit annotation step:

MissingInputException in rule nohit_annotate_reads in file ./KOunt/Snakefile, line 981:
Missing input files for rule nohit_annotate_reads:

Any suggestions on this error?

**Comment by jmattock5** [comment]
Have you downloaded the KOunt reference databases? They are required as input files for this rule so could be the cause if you haven't. If you have please can you check that their names and locations match those in the config.yaml?

**Comment by biojiangke** [comment]
The "KOunt_databases_v1" components all seem to be in the right place. The "KOfam" db is not pointing to in the right directory. I'll update this and give another try. Plus, where do we download the KOfam database file(s)?

**Comment by jmattock5** [comment]
Thank you for spotting that mistake, I've corrected the default path for the KOfam database in the config.yaml. Please try again either with the updated config or it should work if you give your own directory. The KOfam database is downloaded during the kegg_db rule. Once downloaded it isn't overwritten so know you're using the same version of the database for every sample.

**Comment by biojiangke** [comment]
I still got the following error for the test data:

> MissingInputException in rule nohit_annotate_reads in file Snakefile, line 981:
> Missing input files for rule nohit_annotate_reads:
>     output: out/diamond/ERR2027889_005/ERR2027889_005_nohit_besthits, out/Kofamscan/Results/ERR2027889_005/ERR2027889_005_nohit_KOunts

And it seems there is no "out" folder created at this point? Do I need to download a KOfam db and point to a custom location?

**Comment by jmattock5** [comment]
The output directory and KOfam database should both be made by the pipeline as it's running. Please could you try to run the pipeline without the reference databases to help work out where it's breaking?
`snakemake -k --ri --use-conda --cores 8 all_without_reference`

**Comment by biojiangke** [comment]
Thanks for the help! Running without reference successfully finished for the test data set!

**Comment by biojiangke** [comment]
Hi, 

I think I have some clues about the error. After inspecting the input files for this specific step, I noticed that the db input files in the config.ymal are all un-compressed, but by default, the downloaded files in "KOunt_databases_v1" are all "gz" files. I uncompressed the db files and the test commands seem to go through (still running, have not seen the end yet). 

Maybe there are different ways to fix this, like uncompressing the db files after downloading, or configuring the step to run on compressed db files? 

Thanks!
